sdks/java/core/src/main/java/org/apache/beam/sdk/io/UnboundedSource.java

```diff
@@ -76,7 +76,7 @@
    * checkpoint if present.
    */
   public abstract UnboundedReader<OutputT> createReader(
-      PipelineOptions options, @Nullable CheckpointMarkT checkpointMark);
+      PipelineOptions options, @Nullable CheckpointMarkT checkpointMark) throws IOException;
```

Comment:
BoundedSource.createReader throws IOException.
I think we should keep this change in Beam.

If we port this to Dataflow, we can convert IOException to RuntimeException to avoid change the interface.